tests/timeouts.rs

```diff
@@ -1,4 +1,4 @@
-#![cfg(not(target_arch = "wasm32"))]
+#![cfg(not(any(target_arch = "wasm32", feature = "rustls-tls-manual-roots")))]
```

Comment:
Hm, I guess I haven't fully internalized the difference. Looking at the rest of this test file, it seems to just make a "normal" client. Why would that fail?

Is it that if only manual roots are enabled, but none are added, that's an error?

Reply:
> Hm, I guess I haven't fully internalized the difference. Looking at the rest of this test file, it seems to just make a "normal" client. Why would that fail?
> 
> Is it that if only manual roots are enabled, but none are added, that's an error?

Yes, that's the problem.

> And, what if other rustls features are enabled too? It should work, _then_?

Because we can't use the rustls-platform-verifier in conjunction with otherwise-supplied roots on many platforms (only on non-Darwin Unix, see https://github.com/rustls/rustls-platform-verifier/issues/58), this PR chooses to skip gathering roots on all platforms in favor of exclusively using the platform verifier (which comes down to using rustls-native certs on non-Darwin Unix).

Reply:
Yes, as discussed previously in #2159. Do you want to block this addition on https://github.com/rustls/rustls-platform-verifier/issues/58, which would enable making it additive?

Reply:
Thinking from the point of view of users, I believe that is something they would expect to work. So yea, that seems like a requirement.